### PR TITLE
CircleCI update, fix Gem::Package.build invocation

### PR DESCRIPTION
**Pull request by sgerrand**
💁 Some updates:
* upgrades the CircleCI build configuration to use 2.0 format
* runs the test suite in the above against a matrix of Ruby versions (2.3, 2.4, 2.5).
* updates the invocation of `Gem::Package.build` to account for the change in method signature in v3.0.0.

### CODE REVIEW

**Comment by sgerrand** [comment]
💭 I might run RuboCop and Pronto in separate steps from the parameterised RSpec tests.

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+      - publish-gem:
+          filters:
+            branches:
+              ignore: /.*/
+            tags:
+              only: /v[0-9]+(\.[0-9]+){1,2}$/
+          requires:
+            - tests-ruby23
+            - tests-ruby24
+            - tests-ruby25
```
Comment (filipebarros) [inline .circleci/config.yml]: maybe adding `- tests-ruby26` now that ruby 2.6.0 has been released?

Comment (sgerrand) [reply .circleci/config.yml]: 🖖

File: lib/rubygems/versioner/build_plugin.rb [hunk 2]
```diff
@@ -7,9 +9,14 @@ class Package
...
-      def build(spec, skip_validation = false)
+      def build(spec, skip_validation = false, strict_validation = false, file_name = nil)
         spec.version = Gem::Versioner.release_version(spec.version)
-        build_original(spec, skip_validation)
+
+        if Gem::Version.new(Gem::VERSION) >= Gem::Version.new('3.0.0')
+          build_original(spec, skip_validation, strict_validation, file_name)
+        else
+          build_original(spec, skip_validation)
+        end
```
Comment (filipebarros) [inline lib/rubygems/versioner/build_plugin.rb]: 👍 this should work fine in both scenarios now

File: .circleci/config.yml (hunk 1 above)
Comment (sgerrand) [reply .circleci/config.yml]: Done in 2770abb.

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+            - gem-versioner-{{ arch }}-{{ checksum "gem-versioner.gemspec" }}
+      - run:
+          name: Install Ruby dependencies
+          command: bundle check --path=vendor/bundle || bundle install --path=vendor/bundle --jobs=4 --retry=3
+      - save_cache:
+          key: gem-versioner-{{ arch }}-{{ checksum "gem-versioner.gemspec" }}
+          paths:
+            - vendor/bundle
+  pronto:
+    description: Runs RuboCop
```
Comment (bliof) [inline .circleci/config.yml]: Pronto

Comment (sgerrand) [reply .circleci/config.yml]: 🤦‍♂️

Comment (sgerrand) [reply .circleci/config.yml]: Fixed in 233ee53.

File: .circleci/config.yml [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+          name: Build gem
+          command: gem build $CIRCLE_PROJECT_REPONAME.gemspec
+      - run:
+          name: Configure RubyGems API key
+          command: |
+            echo :rubygems_api_key: "$RUBYGEMS_API_KEY" >  ~/.gem/credentials
+            chmod 0600 ~/.gem/credentials
+      - deploy:
+          name: Push gem to RubyGems
+          command: gem push $CIRCLE_PROJECT_REPONAME-*.gem
```
[the remark this answers is not in the document]
Comment (sgerrand) [reply .circleci/config.yml]: 👨‍👧‍👦 👋

**Pull request merged**